### ISSUE TITLE
请问实机飞行需要修改什么参数（What parameters need to be modified for real aircraft flight）

### ISSUE THREAD

**Issue by Mumuyoua** [hku-mars/IPC#2]
我们现在修改了两个参数，一个是悬停油门，一个是允许油门估计。
里程计和地图是fast lio给的
（We have now modified two parameters, one is hover throttle and the other is allow throttle estimation.
Speedometer and map from fast lio）

**Comment by Mumuyoua** [comment]
the flight log I will give later

**Comment by FENYUN323** [comment]
Yes, for other quadrotor platform, you should modify the hover throttle and the flag of throttle estimation these two parameters in `param_real.yaml`.
As an example, we have provided parameter files for actual flight base on MaRS LAB's quadrotor: `param_real.yaml` and `ipc_real.launch`.

**Comment by Mumuyoua** [comment]
According to this code, we estimated that the throttle value should be 0.3, but when we placed the drone there, the actual throttle value turned out to be 0.89. What could be the issue here and how should we modify it? Also, could you please explain the meaning of the highlighted parameters?
[参数.zip](https://github.com/hku-mars/IPC/files/12919323/default.zip)

**Comment by FENYUN323** [comment]
Note that when the quadrotor is not equipped with blades and stationary, the throttle estimation may diverge due to incorrect feedback errors.
Besides, if you already know the hovering throttle, you can turn off the throttle estimate and fill in the throttle estimate in [param_real.yanl](https://github.com/hku-mars/IPC/blob/master/IPC/config/param_real.yaml).
Before entering IPC control, there are the following operations: 
1. Please arming the quadrotor
2. The quadrotor enters off-board mode (PX4 flight controller)
3. IPC enters Command mode (you can modify the joystick definition by entering the RCInCallback function in [planner.cpp](https://github.com/hku-mars/IPC/blob/master/IPC/src/planner.cpp))

**Comment by Mumuyoua** [comment]
> 请注意，当四旋翼飞行器未配备叶片和固定时，由于反馈错误，油门估计可能会发散。此外，如果您已经知道悬停油门，则可以关闭油门估计并在 [param_real.yanl](https://github.com/hku-mars/IPC/blob/master/IPC/config/param_real.yaml) 中填写油门估计。在进入IPC控制之前，有以下几操作：
> 
> 1. 请武装四旋翼
> 2. 四旋翼飞行器进入机外模式（PX4飞控）
> 3. IPC 进入命令模式（您可以通过在 planner 中输入 RCInCallback 函数来修改操纵杆定义[.cpp](https://github.com/hku-mars/IPC/blob/master/IPC/src/planner.cpp))

Thank you for your quick response to our inquiry.
 We have identified the cause of the previous drone crash. Our operational process involved initially taking off in position mode using the internal flight controller to reach a height of approximately 1 meter. We then activated the IPC and waited for the program to start before switching to offboard mode using the remote controller. This approach resulted in the IPC's commands, which were outputted after its activation, not directly affecting the drone's controls and leading to incorrect feedback. However, after reviewing your RC callback function, I noticed that it only has a hover mode and a command control mode. We have another question regarding the operations you mentioned earlier. Should these operations take place after the drone is armed and takes off or after the drone is armed on the ground but before takeoff?

**Comment by FENYUN323** [comment]
Both of the operations you mentioned are fine. Best done after the drone is armed on the ground but before take off.

**Comment by Ma29-HIT** [comment]
> 我们现在修改了两个参数，一个是悬停油门，一个是允许油门估计。 里程计和地图是fast lio给的 （We have now modified two parameters, one is hover throttle and the other is allow throttle estimation. Speedometer and map from fast lio）

Hello, since the code of ROG-Map has not been released, I'm wondering if it can work that running IPC with only FAST-LIO in real.

**Comment by FENYUN323** [comment]
Yes, this version of ipc has a simple map module.

**Comment by Ma29-HIT** [comment]
> Yes, this version of ipc has a simple map module.

Besides, is PX4 1.12 suitable for flight in real? I know that there is some bugs in PX4 1.14. And are there any special params to pay special attention to?

**Comment by YuQ-Zhou** [comment]
@Mumuyoua Hi there, did you completed the real-flight experiments? I have some real-flight experiments I'd like to ask you about. THANKS.
